scripts/configurator/parser/cmake_var.py

```diff
@@ -0,0 +1,39 @@
+def generate_cmake_var(type :str, var):
+   "factory function that associates the correct type of variable"
+   string_to_variable_class = {
+      "BOOL" : cmake_var_bool,
+      "INTERNAL" : cmake_var,
+      "FILEPATH": cmake_var,
+      "STRING" : cmake_var,
+   }
+   variable_instance = string_to_variable_class[type]
+   return variable_instance(var)
+
+
+class cmake_var:
```

Comment:
This should be something like `cmake_var_str`.

Reply:
in place of `var`, right?

Reply:
`cmake_var` should be renamed to `cmake_var_str` or, if you want a generic no-op type, you could name it `cmake_var_generic`.

Reply:
Ok. I think switching names between union type and class name is more appropriate